torch/_dynamo/repro/after_aot.py

```diff
@@ -647,3 +601,277 @@ def inductor_accuracy_fails(fx_g, args, check_str=None):
+                super().__init__(mod)
+                self.subdir = subdir
+
+            def run_node(self, n):
+                r = super().run_node(n)
+                name = n.name
+                if name in known_names:
+                    pbar.update(1)
+                    writer.write_tensor(os.path.join(self.subdir, name), r)
+                return r
+
+        # NB: the module cast doesn't actually do anything, since there are no
+        # parameters/buffers on the module
+        if not options.skip_saving_float64_intermediates:
+            new_mod, new_args = cast_to_fp64(mod, clone_inputs(args))
+            with tqdm(desc="Saving float64 intermediates", total=total) as pbar:
+                WriterInterp(new_mod, "float64").boxed_run(new_args)
+            assert not new_args
+
+        class ExactReaderInterp(fx.Interpreter):
+            def run_node(self, n):
+                r = super().run_node(n)
+                name = n.name
+                if name in known_names:
+                    meta = writer.compute_tensor_metadata(r)
+                    meta2 = reader.read_tensor_metadata(os.path.join("float64", name))
+                    reason = compare_tuples(meta, meta2)
+                    if reason is not None:
+                        pbar.write(f"NONDETERMINISTIC FLOAT64 at {name} ({reason})")
+                    pbar.update(1)
+                return r
+
+        # TODO: check eager determinism
+
+        if not options.skip_check_deterministic:
+            new_mod, new_args = cast_to_fp64(mod, clone_inputs(args))
+            with tqdm(desc="Checking float64 determinism", total=total) as pbar:
+                ExactReaderInterp(new_mod).boxed_run(new_args)
+                assert not new_args
+
+        # Now that we've saved everything, interp through the eager graph
+        # and do comparisons
+        class ReaderInterp(fx.Interpreter):
+            def run_node(self, n):
+                r = super().run_node(n)
+                name = n.name
+                if name in known_names:
+                    inductor = reader.read_tensor(os.path.join("inductor", name))
+                    float64 = reader.read_tensor(os.path.join("float64", name))
+                    logged = False
+
+                    def log_error(msg, *args):
+                        nonlocal logged
+                        logged = True
+                        pbar.write(f"DIVERGED at {name}: {msg % args}")
+
+                    if not same(
+                        r,
+                        inductor,
+                        float64,
+                        tol=torch._dynamo.config.repro_tolerance,
+                        equal_nan=True,
+                        log_error=log_error,
+                    ):
+                        assert logged
+                    pbar.update(1)
+                return r
+
+        with tqdm(desc="Checking divergence", total=total) as pbar:
+            ReaderInterp(mod).boxed_run(args)
```

Comment:
(Another OOM comment) - I suppose this could be run under FakeTensorMode too, so the only memory we use while interpreting the module is when we read into the memory the previously-saved inductor/eager intermediates.

Reply:
It can't actually, because we need the non-float64 eager intermediates